test/framework/src/main/java/org/elasticsearch/cluster/coordination/DeterministicTaskQueue.java

```diff
@@ -430,7 +441,92 @@ public boolean awaitTermination(long timeout, TimeUnit unit) {
 
             @Override
             public ScheduledExecutorService scheduler() {
-                throw new UnsupportedOperationException();
+                return new ScheduledExecutorService() {
```

Comment:
Needed to add a dummy return here since this was used by `org.elasticsearch.index.shard.IndexShard#IndexShard` (only used in the constructor though for the current test, so no actual implementation necessary otherwise).